File: wrappers/dotnet/src/Hyperledger.Ursa.BbsSignatures/Models/VerifyProofRequest.cs
using System;
namespace Hyperledger.Ursa.BbsSignatures
{
    public class VerifyProofRequest
    {
        public VerifyProofRequest(BbsKey publicKey, byte[] proof, IndexedMessage[] messages, string nonce)
        {
            Proof = proof ?? throw new ArgumentNullException(nameof(proof));
            Messages = messages ?? throw new ArgumentNullException(nameof(messages));
            Nonce = nonce ?? throw new ArgumentNullException(nameof(nonce));
            PublicKey = publicKey ?? throw new ArgumentNullException(nameof(publicKey));
        }

        public byte[] Proof { get; set; }

        public IndexedMessage[] Messages { get; set; }

        public string Nonce { get; set; }

        public BbsKey PublicKey { get; set; }
    }
}

File: wrappers/python/ursa_bbs_signatures/models/VerifyProofRequest.py
from typing import List
from .keys import BbsKey


class VerifyProofRequest:
    def __init__(
        self, public_key: BbsKey, proof: bytes, messages: List[str], nonce: bytes
    ) -> None:
        """ Verify Proof Request

        Args:
            public_key: The public key
            proof: The proof
            messages: The messages
            nonce: The nonce
        """
        self.key = public_key
        self.proof = proof
        self.messages = messages
        self.nonce = nonce
